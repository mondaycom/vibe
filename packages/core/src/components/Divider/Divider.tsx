import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
import cx from "classnames";
import React from "react";
import { DirectionType as DirectionTypeEnum } from "./DividerConstants";
import { DividerDirection } from "./Divider.types";
import { getStyle } from "../../helpers/typesciptCssModulesHelper";
import { VibeComponentProps, withStaticPropsWithoutForwardRef } from "../../types";
import styles from "./Divider.module.scss";

export interface DividerProps extends VibeComponentProps {
  /**
   * The direction of the divider.
   */
  direction?: DividerDirection;
  /**
   * If true, removes margin from the divider.
   */
  withoutMargin?: boolean;
}

const Divider = ({
  className = undefined,
  withoutMargin = false,
  direction = "horizontal",
  id,
  "data-testid": dataTestId
}: DividerProps) => {
  return (
    <div
      id={id}
      data-testid={dataTestId || getTestId(ComponentDefaultTestId.DIVIDER, id)}
      className={cx(styles.divider, className, getStyle(styles, direction), {
        [styles.withoutMargin]: withoutMargin
      })}
    />
  );
};

interface DividerStaticProps {
  directions: typeof DirectionTypeEnum;
}

export default withStaticPropsWithoutForwardRef<DividerProps, DividerStaticProps>(Divider, {
  directions: DirectionTypeEnum
});
