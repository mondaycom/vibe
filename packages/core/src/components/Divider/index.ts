export { default as Divider, type DividerProps } from "./Divider";

export * from "./Divider.types";
