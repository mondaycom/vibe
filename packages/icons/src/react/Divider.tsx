/* eslint-disable */
/* tslint:disable */
import * as React from 'react';
export interface DividerProps extends React.SVGAttributes<SVGElement> {
size?: string | number;
}
const Divider: React.FC<DividerProps> = ({size, ...props}) => (
  <svg viewBox="0 0 20 20" fill="currentColor" width={ size || "20" } height={ size || "20" } {...props}>
    <path d="M2.96484 9.57129C2.96484 9.15708 3.30063 8.82129 3.71484 8.82129H16.2863C16.7005 8.82129 17.0363 9.15708 17.0363 9.57129C17.0363 9.9855 16.7005 10.3213 16.2863 10.3213H3.71484C3.30063 10.3213 2.96484 9.9855 2.96484 9.57129Z"
      fill="currentColor" fillRule="evenodd" clipRule="evenodd" />
  </svg>
);
Divider.displayName = 'Divider';
export default Divider;
/* tslint:enable */
/* eslint-enable */
