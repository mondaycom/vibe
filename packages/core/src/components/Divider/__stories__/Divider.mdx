import { Meta } from "@storybook/blocks";
import { CHIP, ICONS, LABEL } from "../../../storybook/components/related-components/component-description-map";
import * as DividerStories from "./Divider.stories";

<Meta of={DividerStories} />

# Divider

- [Overview](#overview)
- [Props](#props)
- [Variants](#variants)
- [Feedback](#feedback)

## Overview

Divider create separation between two UI elements

<Canvas of={DividerStories.Overview} />

### Import

```js
import { Divider } from "@vibe/core";
```

## Props

<PropsTable />

## Variants

### Directions

<Canvas of={DividerStories.Directions} />

## Related components

<RelatedComponents componentsNames={[LABEL, ICONS, CHIP]} />
