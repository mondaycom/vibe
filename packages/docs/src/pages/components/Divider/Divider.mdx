import { Meta } from "@storybook/blocks";
import * as DividerStories from "./Divider.stories";

<Meta of={DividerStories} />

# Divider

Divider create separation between two UI elements

<Canvas of={DividerStories.Overview} />

### Import

```js
import { Divider } from "@vibe/core";
```

## Props

<PropsTable />

## Variants

### Directions

<Canvas of={DividerStories.Directions} />

## Related components

<RelatedComponents componentsNames={["Label", "icons", "Chips"]} />
