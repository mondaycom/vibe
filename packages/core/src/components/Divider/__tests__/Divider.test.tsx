import { it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import Divider from "../Divider";

it("renders correctly with empty props", () => {
  const tree = renderer.create(<Divider />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with custom class name", () => {
  const tree = renderer.create(<Divider className="dummy-class-name" />).toJSON();
  expect(tree).toMatchSnapshot();
});
