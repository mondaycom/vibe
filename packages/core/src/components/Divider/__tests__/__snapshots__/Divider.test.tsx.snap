// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly with custom class name 1`] = `
<div
  className="divider dummy-class-name horizontal"
  data-testid="divider"
/>
`;

exports[`renders correctly with empty props 1`] = `
<div
  className="divider horizontal"
  data-testid="divider"
/>
`;
